### PR TITLE
Initial packaging

### PR DESCRIPTION
**Pull request by kdeldycke**
As discussed in #3, this PR packages the plugin into a proper distributable Python module.

Progress:
* [ ] Rename `master` branch to `main`.
* [ ] Sync repository description with the one from `pyproject.toml`.
* [ ] Setup CI/CD by the way of GitHub workflows.
* [ ] Release v1.0.0 first revision on PyPi.

### CODE REVIEW

**Review by davidlesieur — APPROVED** [review]
Looks good to me! Very good, actually. :wave:
I only left a comment about the readme file.

I have never used Poetry, but this may prompt me to finally start using it!

File: README.md [hunk 1]
```diff
@@ -73,58 +104,25 @@ Perhaps to circumvent this problem, the plugin allows one to manually link
...
 
-Installation
-------------
+To start contributing to this plugin, review the [Contributing to Pelican][] documentation, beginning with the **Contributing Code** section.
 
-To install this plugin, see `How to use plugins
-<http://docs.getpelican.com/en/latest/plugins.html>`__ from the Pelican
-documentation.
+[existing issues]: https://github.com/pelican-plugins/similar-posts/issues
+[Contributing to Pelican]: https://docs.getpelican.com/en/latest/contribute.html
```
[the remark this answers is not in the document]
Comment (kdeldycke) [reply README.md:114]: Let's ask @justinmayer about that. I just copy'n'pasted his template from https://github.com/getpelican/cookiecutter-pelican-plugin/blob/master/%7B%7Bcookiecutter.repo_name%7D%7D/%7B%7Bcookiecutter.readme%7D%7D#L22 😇

**Comment by kdeldycke** [comment]
I propose to move fast and merge this. We'll have plenty of time after that to fix little details. And it will unblock @davidlesieur from iterating and contributing to its own plugin! 😁

**Comment by davidlesieur** [comment]
Merging now would be fine with me! Not sure if @justinmayer wanted to have a look first.

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (justinmayer) [reply README.md:114]: It is indeed intentional. Putting a small blurb in the README is helpful, and the CONTRIBUTING file is needed to replace GitHub's default blurb at the bottom of the new issue form…

> Remember, contributions to this repository should follow our [GitHub Community Guidelines](https://docs.github.com/articles/github-community-guidelines). 

… with …

> Remember, contributions to this repository should follow its [contributing guidelines](https://github.com/pelican-plugins/similar-posts/blob/master/CONTRIBUTING.md).

**Comment by justinmayer** [comment]
Sorry for the delay, folks. I'll have a look at this tomorrow.

**Comment by justinmayer** [comment]
@davidlesieur said:
> I have never used Poetry, but this may prompt me to finally start using it!

For what it's worth, I have been really impressed with Poetry. It's become even more polished with the v1.1 release, and at this point I have no reservations about using it for all my projects.

**Comment by justinmayer** [comment]
Okay, I have reviewed the PR and made some minor changes. Rather than annotate those changes here in the PR, I've taken the liberty of applying and merging them into master. There were simply too many (albeit very minor) bits to annotate, which makes me think that perhaps we should break up these kinds of migrations into multiple pull requests in order to make them more digestible and manageable. 😅 

In short, I think we can close this PR and discuss any follow-up changes in subsequent issues/PRs. If on the other hand there aren't any immediate changes to discuss, then we can publish the first packaged release to PyPI. 🚀 

@kdeldycke: Thank you **very** much for all your work on this endeavor. Much appreciated! 💯

**Comment by kdeldycke** [comment]
@justinmayer no worries! What's important is for that work to reach the main branch so @davidlesieur can start iterating as he see fit.

I'll close this PR then.

**Comment by kdeldycke** [comment]
FYI, I moved all remaining work on a dedicated issue at: https://github.com/pelican-plugins/similar-posts/issues/6

**Comment by justinmayer** [comment]
@kdeldycke: By the way, I made [a few recent Pelican commits](https://github.com/getpelican/pelican/commits), the most recent of which attempts to address the insufficient information regarding how folks should check & correct plugin code style. Thank you for pointing that out in this PR. Much obliged!

**Comment by davidlesieur** [comment]
I was worried by the disappearance of the testing & linting instructions from the README, which can be very useful to newcomers, but I see that all that is well explained in Pelican's docs. Great!

**Comment by justinmayer** [comment]
Originally, the plugin template included all the documentation for setting up one's dev environment, testing, linting, etc. But the feedback we received was that, over time, it would be onerous to update and change that information across dozens of plugin repositories. So we collectively decided to centralize that information in one place, in Pelican's documentation.
